machine_learning/forecasting/run.py

```diff
@@ -0,0 +1,149 @@
+"""
+
+import numpy as np
+import pandas as pd
+from sklearn.preprocessing import Normalizer
+from sklearn.svm import SVR
+from statsmodels.tsa.statespace.sarimax import SARIMAX
+
+
+def lin_reg_pred(train_dt, train_usr, train_mtch, test_dt, test_mtch):
+    """
+    First method: linear regression
+    input : training data (date, total_user, total_event) in list of float
+    output : list of total user prediction in float
+    >>> lin_reg_pred([2,3,4,5], [5,3,4,6], [3,1,2,4], [2,1], [2,2])
+    5.000000000000003
+    """
+    x = []
+    x = [[1, item, train_mtch[i]] for i, item in enumerate(train_dt)]
+    x = np.array(x)
+    y = np.array(train_usr)
+    beta = np.dot(np.dot(np.linalg.inv(np.dot(x.transpose(), x)), x.transpose()), y)
+    prediction = abs(beta[0] + test_dt[0] * beta[1] + test_mtch[0] + beta[2])
+    return prediction
+
+
+def sarimax_predictor(train_user, train_match, test_match):
+    """
+    second method: sarimax
+    sarimax is a statistic method which using previous input
+    and learn its pattern to predict future data
+    input : training data (total_user, with exog data = total_event) in list of float
+    output : list of total user prediction in float
+    >>> sarimax_predictor([4,2,6,8], [3,1,2,4], [2])
+    3.0000424034255513
+    """
+    order = (1, 2, 1)
+    seasonal_order = (1, 1, 0, 7)
+    model = SARIMAX(
+        train_user, exog=train_match, order=order, seasonal_order=seasonal_order
+    )
+    model_fit = model.fit(disp=False, maxiter=600, method="nm")
+    result = model_fit.predict(1, len(test_match), exog=[test_match])
+    return result[0]
+
+
+def support_machine_regressor(x_train, x_test, train_user):
+    """
+    Third method: Support vector regressor
+    svr is quite the same with svm(support vector machine)
+    it uses the same principles as the SVM for classification,
+    with only a few minor differences and the only different is that
+    it suits better for regression purpose
+    input : training data (date, total_user, total_event) in list of float
+    where x = list of set (date and total event)
+    output : list of total user prediction in float
+    >>> support_machine_regressor([[5,2],[1,5],[6,2]], [[3,2]], [2,1,4])
+    1.634932078116079
+    """
+    regressor = SVR(kernel="rbf", C=1, gamma=0.1, epsilon=0.1)
+    regressor.fit(x_train, train_user)
+    y_pred = regressor.predict(x_test)
+    return y_pred[0]
+
+
+def interquartile_range_checker(train_user):
+    """
+    Optional method: interquatile range
+    input : list of total user in float
+    output : low limit of input in float
+    this method can be used to check whether some data is outlier or not
+    >>> interquartile_range_checker([1,2,3,4,5,6,7,8,9,10])
+    2.8
+    """
+    train_user.sort()
+    q1 = np.percentile(train_user, 25)
+    q3 = np.percentile(train_user, 75)
+    iqr = q3 - q1
+    low_lim = q1 - (iqr * 0.1)
+    return low_lim
+
+
+def data_safety_checker(list_vote, actual_result):
```

Comment:
Type hints?

Reply:
arent the hints input and output that i described?? 
or did i do it wrongly?

Reply:
https://docs.python.org/3/library/typing.html is described in CONTRIBUTING.md.  Or look at other Python files in this repo.